src/v/serde/parquet/tests/BUILD

```diff
@@ -0,0 +1,58 @@
+load("@bazel_skylib//rules:run_binary.bzl", "run_binary")
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("//bazel:build.bzl", "redpanda_cc_binary")
+
+redpanda_cc_binary(
+    name = "generate_metadata_binary",
+    srcs = ["generate_metadata.cc"],
+    deps = [
+        "//src/v/base",
+        "//src/v/bytes:iostream",
+        "//src/v/serde/parquet:metadata",
+        "//src/v/serde/parquet:schema",
+        "@seastar",
+    ],
+)
+
+write_file(
+    name = "test_tool_wrapper",
+    out = "test_tool_wrapper.sh",
+    content = [
+        "#!/bin/bash",
+        "exec $@",
+    ],
+)
+
+TEST_CASES = range(4)
+
+[
+    (
+        run_binary(
+            name = "generated_metadata_{}".format(test_case),
+            outs = ["metadata_test_case_{}_output.thrift".format(test_case)],
+            args = [
+                "--output",
+                "$(execpath :metadata_test_case_{}_output.thrift)".format(test_case),
+                "--test-case",
+                "{}".format(test_case),
+            ],
+            tool = ":generate_metadata_binary",
+        ),
+        sh_test(
+            name = "verify_metadata_{}".format(test_case),
+            size = "small",
+            timeout = "short",
+            srcs = [":test_tool_wrapper"],
+            args = [
+                "$(rootpath //src/v/test_utils/go/parquet_metadata_verifier)",
+                "-test-case={}".format(test_case),
+                "-input-file=$(rootpath :generated_metadata_{})".format(test_case),
+            ],
+            data = [
+                ":generated_metadata_{}".format(test_case),
+                "//src/v/test_utils/go/parquet_metadata_verifier",
+            ],
+        ),
+    )
+    for test_case in TEST_CASES
+]
```

Comment:
🔥 

Great to see an example of this. Will be good to do something similar for Iceberg metadata.

Reply:
Happy to help set something up if needed. This is the simplest form I could think of - should be trivial to swap in with python or another language, although golang is really nice because it's dependencies are less annoying generally than python.